x-pack/plugins/canvas/public/components/layout_annotations/border_connection.tsx

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FC } from 'react';
+import PropTypes from 'prop-types';
+import { matrixToCSS } from '../../lib/dom';
+import { TransformMatrix3d } from '../../lib/aeroelastic';
+
+interface Props {
+  height: number;
+  transformMatrix: TransformMatrix3d;
+  width: number;
+}
+
+export const BorderConnection: FC<Props> = ({ transformMatrix, width, height }) => (
```

Comment:
Question: A few of the annotations like this one get called frequently. Should we look into doing any memoization of these components? Or do you think the overhead of the memoization would outweigh any performance gains?

Reply:
That's not a bad idea.  I'm not sure of the perf gain, but the rendering aspect could certainly weigh heavily.  I'll test before committing.